lib/formatter.dart

```diff
@@ -0,0 +1,90 @@
+import 'package:flutter/services.dart';
+
+extension Utilities on String {
+  int getBeginningOfTheLine(int startingIndex) {
+    if (startingIndex <= 0) return 0;
+    for (var i = startingIndex; i >= 0; i--) {
+      if (this[i] == '\n') return i + 1;
+    }
+    return 0;
+  }
+
+  int getEndOfTheLine(int from) {
+    for (var i = from; i < length; i++) {
+      if (this[i] == '\n') return i;
+    }
+
+    return length - 1;
+  }
```

Comment:
i don't understand this sentence

Reply:
You are returning an inclusive index. Meaning, the returned index points to the last character in the line, not to one after that. Usually end indices are represented as exclusive ones. For instance, `'\nSome text'.substring(getBeginOfTheLine(2), getEndOfTheLine(2))` will return `'Some tex'` instead of `'Some text'`